File: backend/src/main/java/com/java/NBE4_5_1_7/domain/news/dto/responseDto/NewResponseDto.java
package com.java.NBE4_5_1_7.domain.news.dto.responseDto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.java.NBE4_5_1_7.domain.news.entity.NewsItem;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Data
@Getter
@Setter
public class NewResponseDto {
    @JsonProperty("lastBuildDate")
    private String lastBuildDate;
    @JsonProperty("total")
    private int total;
    @JsonProperty("start")
    private int start;
    @JsonProperty("display")
    private int display;
    @JsonProperty("items")
    private List<NewsItem> items;
}

File: backend/src/main/java/com/java/NBE4_5_3_7/domain/news/dto/responseDto/NewResponseDto.kt
package com.java.NBE4_5_3_7.domain.news.dto.responseDto

import com.fasterxml.jackson.annotation.JsonProperty
import com.java.NBE4_5_3_7.domain.news.entity.NewsItem

class NewResponseDto(
    @JsonProperty("lastBuildDate") var lastBuildDate: String? = null,
    @JsonProperty("total") var total: Int = 0,
    @JsonProperty("start") var start: Int = 0,
    @JsonProperty("display") var display: Int = 0,
    @JsonProperty("items") var items: List<NewsItem>? = null,
    @JsonProperty("errorMessage") var errorMessage: String? = null
)
